src/org/traccar/protocol/LaipacProtocolDecoder.java

```diff
@@ -45,8 +47,18 @@ public LaipacProtocolDecoder(LaipacProtocol protocol) {
             .number("(d+.d+),")                  // speed
             .number("(d+.d+),")                  // course
             .number("(dd)(dd)(dd),")             // date (ddmmyy)
-            .expression("(.),")                  // type
-            .expression("[^*]+").text("*")
+            .expression("([abZXTSMHFE86430]),")  // event code
+            .number("(d+)").expression("(\\.?)").number("(d*),")    // battery voltage
```

Comment:
Can you please explain logic here.

Reply:
This is the battery voltage. There are two formats for this, you have the old format "3.785" and the new format "3785". They changed the format to not contain the point in the new version. So to support the old an new version, I added an optional point inside the regex.

Reply:
The can be implemented easier. Just remove dots in string and then divide by 1000.